subsys/net/CMakeLists.txt

```diff
@@ -4,6 +4,10 @@ zephyr_library()
 zephyr_library_sources_ifdef(CONFIG_NET_BUF             buf.c)
 zephyr_library_sources_ifdef(CONFIG_NET_HOSTNAME_ENABLE hostname.c)
 
+if(CONFIG_NET_CODE_RELOCATE)
```

Comment:
commit title would to be revised, you are relocating both the net stack and the ethernet L2. And there is no dependency with each other on this. Perhaps splitting the patch in 2 would be clearer: first the net stack, then the ethernet L2.